src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +39,113 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
     return true;
 }
 
+inline int calculate_output_shape(const int& startIdx, const int& stopIdx, const int& step, const int& len) {
+    auto start = (startIdx > len - 1) ? len - 1 : ((startIdx % len + len) % len);
+    auto stop = (stopIdx > len - 1)   ? len     : ((stopIdx % len + len) % len);
+    return (stop - start + step - 1) / step;
```

Comment:
> If step is negative you can get negative dimension is that ok here?
> 
> Check this [sliced dim calc](https://github.com/openvinotoolkit/openvino/blob/b8348cda2e992b2e163a2c326013e59688efbb23/src/core/shape_inference/include/slice_shape_inference_utils.hpp#L172). I think it could be possible to extract this part of code to smaller function and use in template code and here. The `lb` and `ub` is pre-processed start, stop and this part take into account the step direction and avoid using `/` in some cases (but maybe there is possibility to improve it).
> 
> If it is possible to have common code then you get for free extra tests from `type_prop`, because this code is used in `slice` and `strided slice` and tested with dynamic and static shapes.

I compare both functions in performance and coverage. 
- `calculate_output_shape` performs better in most cases but is not faster too much than `get_sliced_value`. 
- `get_sliced_value` seems safe and verified.
- code reuse is better to maintain instead of inventing one.
@praasz @maxnick Do you have other comments for this?

Reply:
I'm not sure  if performance difference between `calculate_output_shape` vs. `get_sliced_valur` can be ignored, but `get_sliced_value` takes into account more corner cases and so I think will be slower always.

If `get_sliced_value` will be little refactored then maybe performance gap can be reduced:
- change parameters to be passed by value.
- the bool flags around lines 143:150 are not always required so could be moved inside `if is_norm_dim_max` as this case I think will be less common. These limit flags are not required, when dimension has no maximum value (or infinite), but it requires refactor if statement for `lb`,`ub` calculation which can be tricky :).

Reply:
> I'm not sure if performance difference between `calculate_output_shape` vs. `get_sliced_valur` can be ignored, but `get_sliced_value` takes into account more corner cases and so I think will be slower always.
> 
> If `get_sliced_value` will be little refactored then maybe performance gap can be reduced:
> 
> * change parameters to be passed by value.
> * the bool flags around lines 143:150 are not always required so could be moved inside `if is_norm_dim_max` as this case I think will be less common. These limit flags are not required, when dimension has no maximum value (or infinite), but it requires refactor if statement for `lb`,`ub` calculation which can be tricky :).

I profile the changes but get very little improvement. And the code becomes not tidy. So I think it is not worth to invest time here.